openmdao/components/exec_comp.py

```diff
@@ -339,6 +341,16 @@ def _setup_expressions(self):
                                        "recognized for variable '%s': %s" %
                                        (self.msginfo, arg, sorted(diff)))
 
+                if 'val' in val and 'value' in val:
+                    raise RuntimeError("Cannot use 'val' and 'value' at the same time, use 'val'.")
+                elif 'value' in val and not warned:
+                    warn_deprecation("'value' will be deprecated in 4.0. Please use 'val' in the "
```

Comment:
I think you need `self.msginfo` in this warning message to point the user to the component in question.
And I think you mean to say that the argument *is* deprecated and will be *removed* in 4.0?
Also, I think you could clean this logic up a bit by merging with the following if statement, e.g.:
```
elif 'value' in val:
    if not warned:
        warn_deprecation(f"{self.msginfo}: the 'value' argument is deprecated and will be removed in version 4.0. Please use 'val'.")
        warned = True
    val['val'] = val.pop('value')
```